### PR TITLE
Feature ver salas disponibles 159956589

### PR DESCRIPTION
**Pull request by joaquinvillar**
**Pivotal:**
https://www.pivotaltracker.com/story/show/159956589

:rotating_light: If you are going to test it remember to add the environment variables!!!!!!!

### CODE REVIEW

**Comment by joaquinvillar** [comment]
I have just added some comments, @Gimli76  tell me if there are okey for you or not.

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -25,12 +25,18 @@ def user_info
...
 
       current_user_info = decoded_token[0]['sub']
-      raise UnauthorizedError.new('Token is valid, but has no subject.', :unauthorized) unless current_user_info
-      raise UnauthorizedError.new("No user_id in token's subject.", :unauthorized) unless current_user_info['user_id']
-      raise UnauthorizedError.new("No email in token's subject.", :unauthorized) unless current_user_info['email']
+      manage_errors(current_user_info)
 
       @current_user_id = current_user_info['user_id']
       @current_user_email = current_user_info['email']
+      ENV['USER_ID'] = @current_user_id.to_s
```
Comment (Gimli76) [inline app/controllers/application_controller.rb]: Where is this environment variable used?
Also, the way I understand the environment (ENV), there is a unique (no concurrent) instance of this variable for the whole system, which is being set for every call to `authorize`. I don't think this is a good idea.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -25,12 +25,18 @@ def user_info
...
+      ENV['USER_EMAIL'] = @current_user_email.to_s
```
Comment (Gimli76) [inline app/controllers/application_controller.rb]: Similar to the idea above. Think of this situation:
User1: calls `authorize` => ENV['USER_EMAIL'] = u1@example.com
User2: calls `authorize` => ENV['USER_EMAIL'] = u2@example.com
User1: calls `google_authentication` = > auth_client.sub = ENV['USER_EMAIL'] = u2@example.com (wrong email!!!)
...

I think you could use parameter in `google_authentication`.

**Comment by LeticiaErrandonea** [comment]
Didn't check the code so I don't know why you needed to set `ENV['USER_ID']` and `ENV['USER_EMAIL']`. I'm just leaving a comment here to say that I totally agree with @Gimli76 and share a [link](https://www.justinweiss.com/articles/how-rails-sessions-work/) on how to use sessions in Rails, that maybe are useful for what you're trying to achieve.

**Comment by Gimli76** [comment]
Thank you for your comment @LeticiaErrandonea!
I would avoid the use of session in this case, though.
Actually, we have been trying to avoid keeping a state on the backend, which is the reason we opted to use JWT for user authorization.

**Comment by joaquinvillar** [comment]
you are right! I didnt see that!

I use that way because I have to access to the email of the logged user in the GoogleConnection class,
How is the right way to access that information? @LeticiaErrandonea @Gimli76

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (joaquinvillar) [reply app/controllers/application_controller.rb]: File reverted! New solution made!!

Thanks!

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (joaquinvillar) [reply app/controllers/application_controller.rb]: you are right I didn't see that!

File: app/controllers/events_controller.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    @event_list = GoogleConnection.get_events(service, filter_date_end, filter_date)
+    render json: events_info(@event_list.items)
+  end
+
+  private
+
+    # filter the useful data
+    def events_info(events)
+      events.map do |event|
+        { title: event.summary, description: event.description, start: event.start.date_time, end: event.end.date_time,\
```
[the remark this answers is not in the document]
Comment (joaquinvillar) [reply app/controllers/events_controller.rb]: Good tip!, didn't know that! I will change it!

**Comment by joaquinvillar** [comment]
I have just notice the team! thanks!

**Pull request merged**